OneCommunityGlobal/HighestGoodNetworkApp
# Jinchao add seeSummaryIndicator and seeVisibilityIcon permissions

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94319381/af4fda89-02d3-4d9b-a662-32344c3268a9)

This PR creates the permissions for seeing weekly summary indicator #880 and visibility icon #859. Now owner can assign these two permissions to user or role to let them view the indicators and icons.

## Main changes explained:
- Create two permissions in `RolePermissions.jsx` and `UserRoleTab.jsx` for both role permissions and user permissions list.
- Replace boolean 'isAdmin' with `hasSummaryIndicatorPermission` and `hasVisibilityIconPermission` in `Leaderboard.jsx`

## How to test:
1. check into current branch
2. log in as owner user (if you don't have one, use my account first and create your own. jinchaoowner@gmail.com password:Ff12345678)
3. Other links->permissions management -> any role. Check if there are two new permissions at the bottom of the role permission list
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94319381/c21bd4e1-682d-4f5e-8154-21d003305368)
4. Assign/unassign these permissions seperately to different roles, check if they are working as intended(don't forget to click save after you make changes):
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94319381/9c2a5e81-ef92-45d2-90cb-6d13729a131d)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94319381/c3331cc1-4337-4f28-aee6-b42904b941c0)
5. Unassign permissions to all roles, then go to Other links->permissions management -> Manage user permissions, check if there are new permissions at the bottom of the permission list.
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94319381/7738a281-4fdd-4202-868e-31090a650830)
6. Assign permissions to one user and log in as that user to check if it works as expected.

## Note:
After changing the permission on roles, it may take more time to reload the page since it changes the `state` of the page.
